evalml/pipelines/utils.py

```diff
@@ -369,6 +386,126 @@ def _make_new_component_name(model_type, component_name, idx=None):
     )
 
 
+def _make_pipeline_from_multiple_graphs(
```

Comment:
Adding this new function but most of it is the same as `_make_stacked_ensemble_pipeline` with some new logic around which final y input to take. I can put up an issue to refactor this moving forward but mainly kept this separate due to the conversations surrounding #2987.

Reply:
@freddyaboulton the implementation for the stacked ensembler assumes that each input pipeline has an estimator so that the final y input chosen to the stacked estimator is the y input going into each pipelines estimator. This usually means that the label encoder or the original y is selected as the final y input. This implementation always chooses the last component that modifies y of the last pipeline. @christopherbunn could give a better explanation of how the stacked implementation works.